src/backups.py

```diff
@@ -107,29 +129,64 @@ def result():
-        """Return the list of backup ids.
+    def _format_backup_list(self, backup_list) -> str:
+        """Formats provided list of backups as a table."""
+        backups = ["{:<21s} | {:<12s} | {:s}".format("backup-id", "backup-type", "backup-status")]
+        backups.append("-" * len(backups[0]))
+        for backup_id, backup_type, backup_status in backup_list:
+            backups.append(
+                "{:<21s} | {:<12s} | {:s}".format(backup_id, backup_type, backup_status)
+            )
+        return "\n".join(backups)
+
+    def _generate_backup_list_output(self) -> str:
+        """Generates a list of backups in a formatted table.
 
-        Args:
-            format_ids: whether to format the ids as (default is False).
+        List contains successful and failed backups in order of ascending time.
         """
-        backup_ids = []
-        _, output, _ = self._execute_command(
-            ["pgbackrest", "repo-ls", f"backup/{self.charm.cluster_name}"]
+        backup_list = []
+        return_code, output, stderr = self._execute_command(
+            ["pgbackrest", "info", "--output=json"]
         )
-        if output:
-            backup_ids = re.findall(r".*[F]$", output, re.MULTILINE)
-            if format_ids:
-                backup_ids = [
-                    datetime.strftime(
-                        datetime.strptime(backup_id[:-1], "%Y%m%d-%H%M%S"), "%Y-%m-%dT%H:%M:%SZ"
-                    )
-                    for backup_id in backup_ids
-                ]
-        return backup_ids
+        if return_code != 0:
+            raise ListBackupsError(f"Failed to list backups with error: {stderr}")
+
+        backups = json.loads(output)[0]["backup"]
+        for backup in backups:
+            backup_id = datetime.strftime(
+                datetime.strptime(backup["label"][:-1], PGBACKREST_BACKUP_ID_FORMAT),
+                BACKUP_ID_FORMAT,
+            )
+            error = backup["error"]
+            backup_status = "finished"
+            if error:
+                backup_status = f"failed: {error}"
+            backup_list.append((backup_id, "physical", backup_status))
+        return self._format_backup_list(backup_list)
+
+    def _list_backups_ids(self) -> List[str]:
+        """Retrieve the list of backup ids.
+
+        Return the list of previously created backups or an empty list if there is no backups
+        in the S3 bucket.
+        """
+        return_code, output, stderr = self._execute_command(
+            ["pgbackrest", "info", "--output=json"]
+        )
+        if return_code != 0:
+            raise ListBackupsError(f"Failed to list backups with error: {stderr}")
+
+        backups = json.loads(output)[0]["backup"]
+        return [
+            datetime.strftime(
+                datetime.strptime(backup["label"][:-1], PGBACKREST_BACKUP_ID_FORMAT),
+                BACKUP_ID_FORMAT,
+            )
+            for backup in backups
```

Comment:
I think we need to add `show_failed: boot` flag to this function.
in one hand, we should return even failed backup id in `_on_create_backup_action`
in another hand we shouldn't try to restore failed backup in `_on_restore_action`
```suggestion
            for backup in backups
            if show_failed or not backup["error"]
```